app/main/api/handlers/index.ts

```diff
@@ -1,4 +1,6 @@
 export { default as ping } from "./ping"
 export { default as nop } from "./nop"
+export { default as echo } from "./echo"
+export { default as error } from "./error"
 
-export type Handler<T> = (system: T, data: any) => Promise<any>
+export type Handler<T> = (system: T, args: Array<any>) => Promise<any>
```

Comment:
The only thing I don't like about it is that all arguments will be of type `any` since I couldn't find an automatic way of validating them before passing them to the handler (I looked into decorators and metadata reflection but that only works for methods, not functions).

My fear is that people writing handlers like:

```typescript
async function translate(system, sentence: string, fromLang: string, toLang: string) { /* ... */ }
```

think that the handler is typesafe, which is not, if the frontend passes a `number` as the `sentence` param that will cause an internal error in the function. Leaving it as `args: Array<any>` in my mind sounds like: you need to validate the args if you want to do anything with them.

Knowing this, what do you think?

Reply:
That makes sense :+1: